firewalldb/privacy_mapper_kvdb.go

```diff
@@ -0,0 +1,222 @@
+package firewalldb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lightninglabs/lightning-terminal/session"
+	"go.etcd.io/bbolt"
+)
+
+/*
+	The PrivacyMapper data is stored in the following structure in the db:
+
+	privacy -> group id -> real-to-pseudo -> {k:v}
+			    -> pseudo-to-real -> {k:v}
+*/
+
+const (
+	txidStringLen = 64
+)
+
+var (
+	privacyBucketKey = []byte("privacy")
+	realToPseudoKey  = []byte("real-to-pseudo")
+	pseudoToRealKey  = []byte("pseudo-to-real")
+
+	pseudoStrAlphabet    = []rune("abcdef0123456789")
+	pseudoStrAlphabetLen = len(pseudoStrAlphabet)
+)
+
+// PrivacyDB constructs a PrivacyMapDB that will be indexed under the given
+// group ID key.
+func (db *BoltDB) PrivacyDB(groupID session.ID) PrivacyMapDB {
+	return &kvdbExecutor[PrivacyMapTx]{
+		db: db.DB,
+		wrapTx: func(tx *bbolt.Tx) PrivacyMapTx {
+			return &privacyMapTx{
+				sessions: db.sessionIDIndex,
+				boltTx:   tx,
+				groupID:  groupID,
+			}
+		},
+	}
+}
+
+// privacyMapTx is an implementation of PrivacyMapTx.
+type privacyMapTx struct {
+	sessions SessionDB
+	groupID  session.ID
+	boltTx   *bbolt.Tx
+}
+
+// asserGroupExists checks that the session group that the privacy mapper is
+// pointing to exists.
+//
+// NOTE: this is technically a DB transaction within another DB transaction.
+// But this is ok because:
+//  1. We only do this for the bbolt backends in which case the transactions are
+//     for _separate_ DB files.
+//  2. The aim is to completely remove this implementation in future.
```

Comment:
Nit: You could also mention that it's acceptable because there's no supported functionality to remove sessions, only to revoke them. Otherwise, that might have caused concurrency issues :)